modules/standard/Random.chpl

```diff
@@ -301,18 +303,19 @@ module Random {
                         else compilerError('choice() size type must be integral or tuple of ranges');
 
       // Return N samples
-      var samples: [1..numElements] A.eltType;
+      var samples: [1..numElements] int;
 
       if replace {
         for sample in samples {
-          var randIdx = stream.getNext(resultType=int, 1, A.size);
-          sample = A[randIdx];
+          var randVal = stream.getNext(resultType=int, 0, X.size-1);
+          var randIdx = X.dim(0).orderToIndex(randVal);
+          sample = randIdx;
         }
       } else {
-        var indices: [A.domain] int = A.domain;
+        var indices: [X] int = X;
```

Comment:
Here is a temporary array we want to avoid creating, because `X` can potentially be very large.